fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -291,19 +294,30 @@ public IndexWriter getWriter(@Nonnull LuceneAnalyzerWrapper analyzerWrapper, @Nu
         return writer;
     }
 
+    void tryClose(Closeable closeable, String what) {
+        try {
+            closeable.close();
+        } catch (Exception ex) {
+            if (LOGGER.isWarnEnabled()) {
+                LOGGER.warn(KeyValueLogMessage.of("close failed for " + what,
+                        LogMessageKeys.AGILITY_CONTEXT, agilityContext), ex);
+            }
+        }
+    }
+
     @Override
     @SuppressWarnings("PMD.CloseResource")
     public synchronized void close() throws IOException {
         if (writer != null) {
-            writer.close();
+            tryClose(writer, "writer");
```

Comment:
This should still throw an exception if any of them fail to close, but it should proceed to the next one. You could use `IOUtils.close(writer, writerReader, directory)` which handles this for you.

Reply:
Nice! 
✅